src/dune_pkg/lock_dir.ml

```diff
@@ -62,30 +78,22 @@ module Source = struct
                 Path.External.relative path source
               else Path.External.of_string source ) )
       ; ( Fields.fetch
-        , fields
-          @@ let+ url = field Fields.url (located string)
-             and+ checksum = field_o Fields.checksum (located string) in
-             let checksum =
-               match checksum with
-               | None -> None
-               | Some ((loc, _) as checksum) -> (
-                 match Checksum.of_string_user_error checksum with
-                 | Ok checksum -> Some (loc, checksum)
-                 | Error e -> raise (User_error.E e))
-             in
-             fun _ -> Fetch { url; checksum } )
+        , let+ fetch = fields decode_fetch in
+          fun _ -> Fetch fetch )
       ]
 
+  let encode_fetch_field { url = _loc, url; checksum } =
```

Comment:
Why is `url` bound once to `_loc` here? The previous code was doing the same, but it doesn't seem to get used.

Reply:
The `url` field is a tuple which is being unpacked. With parens it would be `{ url = (_loc, url); ... `.

Reply:
Ah! I totally missed the `,`, thanks.